src/aiida/tools/dumping/processes.py

```diff
@@ -0,0 +1,454 @@
+                    f'Not removing for safety reasons.'
+                )
+
+        # Not included in if-else as to avoid having to repeat the `mkdir` call.
+        # `exist_ok=True` as checks implemented above
+        validate_path.mkdir(exist_ok=True, parents=True)
+
+        return validate_path.resolve()
+
+    def _generate_calculation_io_mapping(self, io_dump_paths: Optional[List[Any]] = None) -> SimpleNamespace:
+        """Helper function to generate mapping for entities dumped for each `CalculationNode`.
+
+        This is to avoid exposing AiiDA terminology, like `repository` to the user, while keeping track of which
+        entities should be dumped into which directory, and allowing for alternative directory names.
+
+        :param io_dump_paths: Subdirectories created for the `CalculationNode`.
+            Default: ['inputs', 'outputs', 'node_inputs', 'node_outputs']
+        :return: SimpleNamespace mapping.
+        """
+
+        aiida_entities_to_dump = ['repository', 'retrieved', 'inputs', 'outputs']
+        default_calculation_io_dump_paths = ['inputs', 'outputs', 'node_inputs', 'node_outputs']
+        empty_calculation_io_dump_paths = [''] * 4
+
+        if self.flat and io_dump_paths is None:
+            LOGGER.info(
+                'Flat set to True and no `io_dump_paths`. Dumping in a flat directory, files might be overwritten.'
+            )
+            return SimpleNamespace(**dict(zip(aiida_entities_to_dump, empty_calculation_io_dump_paths)))
+
+        elif not self.flat and io_dump_paths is None:
+            LOGGER.info(
+                'Flat set to False but no `io_dump_paths` provided. '
+                + f'Will use the defaults {default_calculation_io_dump_paths}.'
+            )
+            return SimpleNamespace(**dict(zip(aiida_entities_to_dump, default_calculation_io_dump_paths)))
+
+        elif self.flat and io_dump_paths is not None:
+            LOGGER.info('Flat set to True but `io_dump_paths` provided. These will be used, but `inputs` not nested.')
+            return SimpleNamespace(**dict(zip(aiida_entities_to_dump, io_dump_paths)))
+        else:
+            LOGGER.info(
+                'Flat set to False but no `io_dump_paths` provided. These will be used, but `node_inputs` flattened.'
+            )
+            return SimpleNamespace(**dict(zip(aiida_entities_to_dump, io_dump_paths)))  # type: ignore[arg-type]
+
+    def _dump_node_yaml(
+        self,
+        process_node: ProcessNode,
+        output_path: Path,
+        output_filename: str = '.aiida_node_metadata.yaml',
+    ) -> None:
+        """Dump the selected `ProcessNode` properties, attributes, and extras to a YAML file.
+
+        :param process_node: The `ProcessNode` to dump.
+        :param output_path: The path to the directory where the YAML file will be saved.
+        :param output_filename: The name of the output YAML file. Defaults to `.aiida_node_metadata.yaml`.
+        """
+
+        _node_properties = [
```

Comment:
I added the underscore to show that it's a hard-coded, function-internal variable that shouldn't be touched by the user. As there's no way to modify it via the Python API or `verdi` CLI anyway, this might be redundant. Not sure what the best practice here is, but for me, it's also fine to remove the underscore, but I don't have a strong opinion. Whatever follows best coding practices, @sphuber?

Reply:
I wouldn't say it is very critical, but I would side with @agoscinski that given the fact that it is a locally scoped variable, there is node point in underscoring. So I would remove it.